xds/internal/balancer/ringhash/config.go

```diff
@@ -49,6 +51,12 @@ func parseConfig(c json.RawMessage) (*LBConfig, error) {
 	if cfg.MaxRingSize == 0 {
 		cfg.MaxRingSize = defaultMaxSize
 	}
+	if cfg.MinRingSize > maxRingSizeCap {
+		cfg.MinRingSize = maxRingSizeCap
+	}
+	if cfg.MaxRingSize > maxRingSizeCap {
+		cfg.MaxRingSize = maxRingSizeCap
+	}
 	if cfg.MinRingSize > cfg.MaxRingSize {
 		return nil, fmt.Errorf("min %v is greater than max %v", cfg.MinRingSize, cfg.MaxRingSize)
 	}
```

Comment:
Should this check be done first?  Otherwise it passes if Min > Max > MaxCap.  Or is that okay?

Reply:
Because MaxCap is supposed to clamp min and max, I don't see a benefit of failing (beyond catching weird configuration). The client is still able to make use of such data sensibly, because of the clamp, so it *seems* overly fragile to fail.